src/sagemaker_containers/entry_point.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License'). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the 'license' file accompanying this file. This file is
+# distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import enum
+import os
+import sys
+
+from sagemaker_containers import _env, _errors, _files, _logging, _modules, _process
+
+
+def run(uri, user_entry_point, args, env_vars=None, wait=True):
+    # type: (str, str, list, dict, bool) -> subprocess.Popen
+    """Runs the user entry point, passing env_vars as environment variables and args as command arguments.
```

Comment:
This function does more than running the entry point. It downloads the source files, sometimes install it as a model and sets up the environment variables. Please add a bit more explanation here.